x-pack/plugins/observability_ai_assistant/public/hooks/use_timeline.ts

```diff
@@ -116,31 +125,35 @@ export function useTimeline({
         return nextSubscription;
       });
     }).then(async (reply) => {
-      if (reply.error) {
+      if (reply?.error) {
         return nextMessages;
       }
-      if (reply.aborted) {
+      if (reply?.aborted) {
         return nextMessages;
       }
 
       setPendingMessage(undefined);
 
-      const messagesAfterChat = nextMessages.concat({
-        '@timestamp': new Date().toISOString(),
-        message: {
-          ...reply.message,
-        },
-      });
+      const messagesAfterChat = reply
+        ? nextMessages.concat({
+            '@timestamp': new Date().toISOString(),
+            message: {
+              ...reply.message,
+            },
+          })
+        : nextMessages;
 
       onChatUpdate(messagesAfterChat);
 
-      if (reply?.message.function_call?.name) {
-        const name = reply.message.function_call.name;
+      const lastMessage = last(messagesAfterChat);
```

Comment:
instead of using reply, we pick the last item from messages, so we can execute a function suggestion if it has been edited by the user.